src/sysinfo

```diff
@@ -558,16 +563,49 @@ function get_admin_nic_tag()
     eval "Admin_NIC_Tag=${fields[1]:-"admin"}"
 }
 
+# Find one IP address associated with a NIC tag by parsing nictagadm and
+# ifconfig output.
+function get_tag_ip()
+{
+    local tag=$1
+    local fields nic
+
+    nic=$(/usr/bin/nictagadm list -p | while IFS=: read -a fields; do
+        if [[ "${fields[0]}" == "$tag" ]]; then
+            echo "${fields[2]}"
+            break
+        fi
+    done)
+    if [[ -z "$nic" ]]; then
+        return
+    fi
+
+    /usr/sbin/ifconfig "$nic" | while read -a fields; do
```

Comment:
I'd suggest using ipadm instead of parsing the ifconfig output -- AFAIK it should have the same info here.

Reply:
The problem with using ipadm is that we don't know the addrobj reference. With static networking it'll be `${nic}/_a` (usually? the `_X` *can* be different, but on SmartOS it's probably always going to be `_a`), but with dhcp it's going to be `${nic}/?`.

Generally I'd be inclined to use ipadm as well, but I'm just following up with this PR, hoping to get it merged, and ifconfig is what @mgerdts had used. Given what we know, ifconfig is probably the only choice currently.